torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2502,7 +2331,221 @@ void ProcessGroupNCCL::watchdogHandler() {
       // in case processing is slowed down (but not hung) by cuda api contention
       heartbeat_++;
     }
-    done = workMetaList_.empty();
+    done = pg_->workMetaList_.empty();
+  }
+}
+
+uint64_t ProcessGroupNCCL::Watchdog::getHeartbt() const {
+  return heartbeat_.load();
+}
+
+void ProcessGroupNCCL::Watchdog::setDesyncDebug(bool desyncDebug) {
+  desyncDebug_ = desyncDebug;
+}
+
+// Initialize and enable DesyncDebugger
```

Comment:
From this Line to L2548 is just move code around so that all watchdog code stay together.

Reply:
Can you please minimize code change in this PR?